src/main/java/Deadline.java

```diff
@@ -0,0 +1,36 @@
+import java.time.LocalDate;
+import java.time.LocalTime;
+import java.time.format.DateTimeFormatter;
+
+/**
+ * Deadline Command to add in an Deadline type Task.
+ */
+public class Deadline extends Task {
+    LocalDate date;
+    LocalTime time;
+    String preposition;
+
+    Deadline(String title, String preposition, LocalDate date, LocalTime time)
+            throws WrongUsageException {
+        super(title);
+        this.name = "deadline";
+        this.usage = "deadline [TaskToBeDone] /by [DD/MM/YYYY] [HH:MM]";
+        this.description = "Stores a task in the list marked as a deadline";
+        if (title.isEmpty() || preposition.isEmpty()) {
+            throw new WrongUsageException(this.name, this.usage);
+        }
+        this.preposition = preposition;
+        this.date = date;
+        this.time = time;
+        this.saveRep = "[D] " + super.toString() + " (" + preposition + ": " +
+                date + " " +
+                time + ")";
```

Comment:
Perhaps there is no need to split this into 2 lines? May be easier for the reader to read that way.